### PR TITLE
Make reshard work with streams too

### PR DESCRIPTION
**Pull request by akoshelev**
Internally, reshard used streams already, so it is only a matter of changing the API and connecting things together

### CODE REVIEW

**Comment by akoshelev** [comment]
@eriktaubeneck This may be what you want. This PR is not complete (I just quickly draw an API change), so we can't merge it, but the only thing that is missing is test coverage.

**Comment by eriktaubeneck** [comment]
test added @akoshelev

File: ipa-core/src/protocol/context/mod.rs
Comment (akoshelev) [inline ipa-core/src/protocol/context/mod.rs:868]: the comment needs to be changed, otherwise lgtm

Comment (akoshelev) [reply ipa-core/src/protocol/context/mod.rs:868]: heh, I can't approve my own PRs

**Review by eriktaubeneck — APPROVED** [review]
great work, thanks for unblocking!

File: ipa-core/src/protocol/context/mod.rs
Comment (akoshelev) [reply ipa-core/src/protocol/context/mod.rs:868]: I thought the purpose of this test is to ensure we can use stream API, ensuring global record order in two tests seems redundant. But that's a nit, lets merge it and follow up if needed

Comment (eriktaubeneck) [reply ipa-core/src/protocol/context/mod.rs:868]: ahh I see you're point. I can open a small PR to clarify that. we want global ordering, and we want that to work with both APIs

**Pull request merged**
